src/zeroD/Reactor.cpp

```diff
@@ -445,20 +444,25 @@ void Reactor::setAdvanceLimits(const double *limits)
         throw CanteraError("getState",
                            "Error: reactor is empty.");
     }
-    for (size_t j = 0; j < m_nv; j++) {
-        m_advancelimits[j] = limits[j];
+    m_advancelimits.resize(m_nv, -1.0);
+    std::copy(limits, limits + m_nv, m_advancelimits.begin());
+
+    // resize to zero length if no limits are set
+    if (std::none_of(m_advancelimits.begin(), m_advancelimits.end(),
+                     [](double l){return l>0;})) {
```

Comment:
Can we have a variable name besides `l`? Looks way too much like `1` in many fonts. Applies to both instances of this.